library/core/src/option.rs

```diff
@@ -559,6 +559,38 @@ impl<T> Option<T> {
         }
     }
 
+    /////////////////////////////////////////////////////////////////////////
+    // Setting a new value
+    /////////////////////////////////////////////////////////////////////////
+
+    /// Inserts `value` into the option then returns a mutable reference to it.
+    ///
```

Comment:
Might be good to add a note about what happens with the old value. Maybe something like:

```suggestion
    ///
    /// If the option already contains a value, the old value is dropped.
    ///
```